### ISSUE TITLE
[Bug] nnpackage_run exits with assert on Gather

### ISSUE THREAD

**Issue by chunseoklee** [Samsung/ONE#5774]
Since a0a92f07, nnpackage_run fails to run model [mobilebert_2gather.zip](https://github.com/Samsung/ONE/files/5848187/mobilebert_2gather.zip) : 

```
ubuntu@ubuntu ~> Product/out/bin/nnpackage_run mobilebert_2gather/
Package Filename mobilebert_2gather/
nnpackage_run: /home/twoflower/dev/ONE/compute/cker/include/cker/operation/Gather.h:62: void nnfw::cker::Gather(const nnfw::cker::GatherParams&, const nnfw::cker::Shape&, const T*, const nnfw::cker::Shape&, const CoordsT*, const nnfw::cker::Shape&, T*) [with T = float; CoordsT = int]: Assertion `coords_data[i] < axis_size' failed.
fish: 'Product/out/bin/nnpackage_run m…' terminated by signal SIGABRT (Abort)
``` 

Note that I have tested this on ubuntu 18.04 armv7l(rpi4). 
The attached model is extracted from https://tfhub.dev/tensorflow/lite-model/mobilebert/1/default/1 (2nd node Gather)

(updated 2021-01-29 : based on https://github.com/Samsung/ONE/issues/5774#issuecomment-765193513, we found that https://github.com/Samsung/ONE/commit/a0a92f07e90d9835ba2a1638d268e3c95abbd60a induces dynamic_cast issue)

**Comment by chunseoklee** [comment]
cc @wateret

**Comment by wateret** [comment]
I'll take a look soon.

**Comment by wateret** [comment]
In my environment,

f4a1758c57f036f3632714bbdb5df84b2b480f3a (current master) Debug : ASSERTION ERROR
f4a1758c57f036f3632714bbdb5df84b2b480f3a (current master) Release : NO ERROR
a0a92f0 Debug : ASSERTION ERROR
a0a92f0 Release : NO ERROR

Same results for both commits, so I'm not sure if that commit is the real issue.

**Comment by chunseoklee** [comment]
> Same results for both commits, so I'm not sure if that commit is the real issue.

I will check again soon.

**Comment by wateret** [comment]
FYI my XU4 is ubuntu 16.04.

**Comment by chunseoklee** [comment]
I have found : 
a0a92f0 Release on android : segfault
a0a92f0 Debug on android : 
```
 adb shell LD_LIBRARY_PATH=/data/local/tmp/Product/out/lib /data/local/tmp/Product/out/bin/nnpackage_run /data/local/tmp/mobilebert_2gather -w 1 -r 4 -m 1
Package Filename /data/local/tmp/mobilebert_2gather
/home/twoflower/dev/ONE/runtime/onert/core/include/backend/cpu_common/BackendContextHelpers.h:266: void onert::backend::cpu_common::initConsts(onert::backend::BackendContext &): assertion "ext_tensor" failed
Aborted (core dumped)
``` 

Note that rpi4 shows the same reesult of yours. Thus, segfault is reproduced on android device.

**Comment by wateret** [comment]
In a0a92f0 , runtime/onert/core/include/backend/cpu_common/BackendContextHelpers.h:266

It is
```
    ExternalTensor *ext_tensor = dynamic_cast<ExternalTensor *>(tensor);
    assert(ext_tensor)
```

And this may have something to do with "key function" issue.

**Comment by dayo09** [comment]
@chunseoklee By the way, this isn't android build, so cannot be a ndk issue. 😕 ❓

**Comment by chunseoklee** [comment]
> @chunseoklee By the way, this isn't android build, so cannot be a ndk issue. 😕 ❓

Issue has been reported based on result from debug binary for RPI devices. 
However, it turns out that **release** binary for android produces segfault ( https://github.com/Samsung/ONE/issues/5774#issuecomment-765150071 ). Thus, ndk issue.

**Comment by hseok-oh** [comment]
Resolved by #5898
